ddtrace/profiling/collector/_traceback.pyx

```diff
@@ -51,9 +92,18 @@ cpdef pyframe_to_frames(frame, max_nframes):
     nframes = 0
     while frame is not None:
         nframes += 1
-        if len(frames) < max_nframes:
-            code = frame.f_code
-            lineno = 0 if frame.f_lineno is None else frame.f_lineno
-            frames.append((code.co_filename, lineno, code.co_name, _extract_class_name(frame)))
-        frame = frame.f_back
+        # Python 3.11 moved PyFrameObject to internal C API and cannot be directly accessed from tstate
+        IF PY_MAJOR_VERSION > 3 or (PY_MAJOR_VERSION == 3 and PY_MINOR_VERSION >= 11):
+            if len(frames) < max_nframes:
+                code = PyFrame_GetCode(<PyFrameObject*> frame)
+                lineno = PyFrame_GetLineNumber(<PyFrameObject*> frame)
+                lineno = 0 if lineno is None else lineno
+                frames.append(((<object>code).co_filename, lineno, (<object>code).co_name, _extract_class_name(frame)))
+            frame = <object>PyFrame_GetBack(<PyFrameObject*> frame)
```

Comment:
`PyFrame_GetBack()` returns a strong reference. Should I be calling `Py_XDECREF(frame)` even though we are iterating through these references?

Reply:
the frame isn't actually needed anymore after L114 right? since we are creating/returning a list of tuples?

does the frame getting GC'd cause co_filename/co_name/etc to also get GC'd? are we appropriately incref/decref-ing those references?